Stage 1
Project 1: BASh Basic
You are to achieve this short story with the command line alone. 
Create your copy of the file and enter your command in the terminal space ($) below each action.
Your Team Name: 
Participants who contributed significantly (slack handle alone): 

N/B: The story here is fictional and the files are just hypothetical. Please don’t use it for any serious research work. 
Please copy exactly what worked. Do not paraphrase. A single mismatch makes you loose your point.
To submit this project, make this document open using the 🔒share icon at the top right corner. Copy the link and submit it on HackBio platform (See submission Guide).

Login to your coding workspace
Create a folder titled your name

$ 


Create another new directory titled biocomputing and change to that directory with one line of command

$ 


Download these 3 files:
https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.fna 
https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk
https://raw.githubusercontent.com/josoga2/dataset-repos/main/wildtype.gbk 

$ 


OH! You made a mistake. You have to move the .fna file to the folder titled your name directly. (Do this with one command. Hint: See our cheatsheet)

$ 


OH No! The gbk file is a duplicate, they are actually the same thing. Please delete it.

$ 


The .fna file is actually from a bacteria, and it should definitely have a TATA (tata) box for initiating gene transcription. The molecular biologist is trying to understand the implication of dual TATA sequences. The files got mixed up and we are not sure which is wildtype and which is mutant. The mutant should have “tatatata” while the normal should have just “tata”. Can you confirm if the file is mutant or wild type

$ 


If it is mutant, print all the lines that show it is a mutant into a new file

$ 


What is your favorite gene? (In any organism). Each team member should pick a unique gene different from every other person 

$ 


Download the fasta format of the gene from NCBI Nucleotide

$ 


How many lines are in the FASTA file (with the exception of the header)

$ 


How many times does A occur

$ 


How many times does G occur

$ 


How many times does C occur

$ 


How many times does T occur

$ 


Calculate the %GC content of your gene

$ 


Create a nucleotide (.fasta) file title your name

$ 


“echo” the following into the file using >>: the number of A, G, T and C in the file you created above.

$ 


Upload the file to your team’s github repo in a folder called /output

$ link here


Save all the codes you have used in this project in a file named yourname.sh Upload all the codes you have used to your team’s github repo in a folder called /script

$ link here


Clear your terminal space and print all the commands you have used today.

$ 


List the files in the two folders and share a screenshot of your terminal below

$ 


Take a screenshot of your terminal screen currently and paste it below

Project 2: Installing Bioinformatics Softwares on the terminal
 N/B: You need to install and setup your conda environment with either anaconda or miniconda.
Please copy exactly what worked. Do not paraphrase. A single mismatch makes you loose your point.

Activate your base conda environment

$ 


Create a conda environment names funtools

$ 


Activate the funtools environment

$ 


Install Figlet using conda

$ 


Run the following command figlet {your name}. Put a screenshot of what you see below 😀

Install bwa through the bioconda channel

$ 


Install blast through the bioconda channel

$ 


Install samtools through the bioconda channel

$ 


Install bedtools through the bioconda channel

$ 


Install spades.py through the bioconda channel

$ 


Install bcftools through the bioconda channel

$ 


Install fastp through the bioconda channel

$ 


Install multiqc through the bioconda channel

$ 



To submit this project, make this document open using the 🔒share icon at the top right corner. Copy the link and submit it on HackBio platform.

Finally, everyone in your team should be ready to discuss your code submission with everyone.
Learning Resources:
The Official learning resource for this internship is HackBio’s Genomics Course. Sign up to enjoy uninterrupted and synchronized flow of bioinformatics knowledge. If you have access to the course already, everything you need for the internship is already provided in the course.
However, we have plans for you if you are unable to purchase the course. We have gathered some resources for you to help you learn and navigate the internship better.
Stage 1
How to Access the terminal for the purpose of stage 1 and 2
Introduction to BASh
How to setup and use Conda
An article on bioconda usage from HackBio
The rest is practice! practice!! practice!!!
 
